dev/fattest.simplicity/src/com/ibm/websphere/simplicity/beansxml/BeansAsset.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.websphere.simplicity.beansxml;
+
+import org.jboss.shrinkwrap.api.asset.ClassLoaderAsset;
+import org.jboss.shrinkwrap.impl.base.asset.AssetUtil;
+
+/**
+ * Shrinkwrap asset for a beans.xml file
+ * <p>
+ * Example usage, either using CDIArchiveHelper...
+ *
+ * <pre>
+ * <code>
+ * WebArchive war = ShrinkWrap.create(WebArchive.class)
+ *              .addPackage(MyClass.class.getPackage());
+ *
+ * war = CDIArchiveHelper.addBeansXML(war, Mode.ALL);
+ * </code>
+ * </pre>
+ *
+ * ...or directly...
+ *
+ * <pre>
+ * <code>
+ * BeansAsset beans = new BeansAsset(Mode.ALL);
+ *
+ * WebArchive war = ShrinkWrap.create(WebArchive.class)
+ *              .addPackage(MyClass.class.getPackage())
+ *              .addAsWebInfResource(beans, "beans.xml");
```

Comment:
A static constructor method may make this nicer if you were actually to use it this way. Seems odd to be calling a constructor to get what is effectively a constant.

```
 * WebArchive war = ShrinkWrap.create(WebArchive.class)
 *              .addPackage(MyClass.class.getPackage())
 *              .addAsWebInfResource(getBeansXml(ALL), "beans.xml");
```

Reply:
OK I added a static constructor method and cached the possible results.